### PR TITLE
439: tokenFallback changes in feeHandler

### PR DESCRIPTION
**Pull request by Deceivious**
Added index to store current index
Added method to add allowed addresses
Added readonly method to view allowed addresses
Added method to deligate fees to other contracts
Added unittests

### CODE REVIEW

File: core_contracts/feehandler/feehandler.py [hunk 1]
```diff
@@ -157,36 +158,82 @@ def tokenFallback(self, _from: Address, _value: int, _data: bytes) -> None:
...
+        self._current_allowed_addresses_index.set(current_index)
+
+        try:
+            # Raises JSONDecodeError if trying to decode an empty string.
+            path = json_loads(self._routes[self.msg.sender][self._getContractAddress("baln")])
+        except:
+            path = []
+
+        # Attempt to process fees. Catch Error and report via eventlog if failure.
+        balance = self._getTokenBalance(address)
```
Comment (dbrehmer) [inline core_contracts/feehandler/feehandler.py]: Repeat of line 210.

File: core_contracts/feehandler/feehandler.py [hunk 2]
```diff
@@ -157,36 +158,82 @@ def tokenFallback(self, _from: Address, _value: int, _data: bytes) -> None:
...
+
+            else:
+                # Use dex.
+                self._transferToken(address, self._getContractAddress("dex"), balance,
+                                    self._createDataFieldDex(self._getContractAddress("baln"),
+                                                             self._getContractAddress("dividends")))
+
+        except BaseException as e:
+            revert(f'Fee conversion for {address} failed, {repr(e)}')
+            # self.FeeNotProcessed(self.msg.sender, repr(e))
```
Comment (dbrehmer) [inline core_contracts/feehandler/feehandler.py]: Looks like we will not be needing this eventlog. Let's remove it, along with the eventlog comment on line 219 and the eventlog definition at the end.

**Review by dbrehmer — CHANGES_REQUESTED** [review]
Code looks great, thanks. Just a couple small things to clean up.

File: core_contracts/feehandler/feehandler.py (hunk 1 above)
Comment (Deceivious) [reply core_contracts/feehandler/feehandler.py]: Code @ 219 and 220 removed.

File: core_contracts/feehandler/feehandler.py (hunk 2 above)
Comment (Deceivious) [reply core_contracts/feehandler/feehandler.py]: Commented code removed along with the definition

**Review by dbrehmer — APPROVED** [review]
Looks good.

**Pull request merged**